packages/neutron-sdk/src/interchain_queries/types.rs

```diff
@@ -187,6 +210,259 @@ impl KVReconstruct for Balances {
+    /// status is the validator status (bonded/unbonding/unbonded).
+    pub status: i32,
+    /// tokens define the delegated tokens (incl. self-delegation).
+    pub tokens: String,
+    /// delegator_shares defines total shares issued to a validator's delegators.
+    pub delegator_shares: String,
+    /// moniker defines a human-readable name for the validator.
+    pub moniker: Option<String>,
+    /// identity defines an optional identity signature (ex. UPort or Keybase).
+    pub identity: Option<String>,
+    /// website defines an optional website link.
+    pub website: Option<String>,
+    /// security_contact defines an optional email for security contact.
+    pub security_contact: Option<String>,
+    /// details define other optional details.
+    pub details: Option<String>,
+    /// unbonding_height defines, if unbonding, the height at which this validator has begun unbonding.
+    pub unbonding_height: u64,
+    /// unbonding_time defines, if unbonding, the min time for the validator to complete unbonding.
+    pub unbonding_time: Option<u64>,
+    /// rate is the commission rate charged to delegators, as a fraction.
+    pub rate: Option<Decimal>,
+    /// max_rate defines the maximum commission rate which validator can ever charge, as a fraction.
+    pub max_rate: Option<Decimal>,
+    /// max_change_rate defines the maximum daily increase of the validator commission, as a fraction.
+    pub max_change_rate: Option<Decimal>,
+    /// update_time is the last time the commission rate was changed.
+    pub update_time: Option<u64>,
+    /// min_self_delegation is the validator's self declared minimum self delegation.
+    pub min_self_delegation: Decimal,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+/// A structure that can be reconstructed from **StorageValues**'s for the **Balance Interchain Query**.
+/// Contains validator info from remote chain.
+pub struct StakingValidator {
+    pub validator: Option<Validator>,
+}
+
+impl KVReconstruct for StakingValidator {
+    fn reconstruct(storage_values: &[StorageValue]) -> NeutronResult<StakingValidator> {
+        if let Some(kv) = storage_values.iter().next() {
+            let validator: CosmosValidator = CosmosValidator::decode(kv.value.as_slice())?;
+            let description = &validator.description;
+            let commission = &validator.commission;
+
+            let validator = Validator {
+                operator_address: validator.operator_address,
+                delegator_shares: validator.delegator_shares,
+                jailed: validator.jailed,
+                status: validator.status,
+                tokens: validator.tokens,
+                unbonding_height: validator.unbonding_height as u64,
+                unbonding_time: if validator.unbonding_time.is_none() {
+                    None
+                } else {
+                    Some((validator.unbonding_time.unwrap().seconds) as u64)
+                },
```

Comment:
You can do such `if` using `map`, which requires only one line of code:

```suggestion
                unbonding_time: validator.unbonding_time.map(|v| v.seconds as u64),
```

Do this for all such similar places here